### PR TITLE
Local signin

### PR DESCRIPTION
**Pull request by KyuaKwon**
Hi new TL!

I updated user.controller.js and unit test of it.
Now we check the validation of secret code, so only the lastest one would work.

### CODE REVIEW

File: src/server/controllers/users.controller.js [hunk 1]
```diff
@@ -163,32 +165,49 @@ export function localSignIn(req, res, next) {
...
-  User.findOne({ email: req.body.email }).exec()
-    .then(user => {
-      if (!user) {
-        throw new Error(userCallback.ERR_USER_NOT_FOUND);
-      } else {
-        // TODO: Save secret code to db and check validaion of it. Only the last one is valid.
-        mailingController.sendEmail(req.body.email, mailStrings.RESETPW_SUBJECT,
-          mailStrings.RESETPW_HTML, secretCode);
+  if (req.body.email) {
+    User.findOne({ email: req.body.email }).exec()
```
Comment (SarahhhC) [inline src/server/controllers/users.controller.js]: Let's unify the form, when find a user, use `_id`.

File: src/server/controllers/users.controller.js [hunk 2]
```diff
@@ -163,32 +165,49 @@ export function localSignIn(req, res, next) {
...
+              let secretCode = new SecretCode();
+              secretCode.email = req.body.email;
+              secretCode.secretCode
+                = cipher.update(new Date().toISOString(), 'utf-8', 'hex') + cipher.final('hex');
+              return secretCode.save();
+            })
+            .then(secretCode => {
+              mailingController.sendEmail(req.body.email, mailStrings.RESETPW_SUBJECT,
+                mailStrings.RESETPW_HTML, secretCode.secretCode);
+              res.status(201).json({ secretCode: secretCode.secretCode });
```
Comment (SarahhhC) [inline src/server/controllers/users.controller.js]: Why status is 201??

File: src/server/controllers/users.controller.js [hunk 3]
```diff
@@ -163,32 +165,49 @@ export function localSignIn(req, res, next) {
...
-        mailingController.sendEmail(req.body.email, mailStrings.RESETPW_SUBJECT,
-          mailStrings.RESETPW_HTML, secretCode);
+  if (req.body.email) {
+    User.findOne({ email: req.body.email }).exec()
+      .then(user => {
+        if (!user) {
+          throw new Error(userCallback.ERR_USER_NOT_FOUND);
+        } else {
+          SecretCode.findOne({ email: req.body.email, isValid: true }).exec()
+            .then(validSecretCode => {
```
Comment (SarahhhC) [inline src/server/controllers/users.controller.js]: If you return `SecretCode.findOne({ email: req.body.email, isValid: true }).exec()` here, 
you can use non-nested promise!

Comment (KyuaKwon) [inline src/server/controllers/users.controller.js]: updated!

File: src/server/controllers/users.controller.js (hunk 2 above)
Comment (KyuaKwon) [inline src/server/controllers/users.controller.js]: updated!

**Comment by SarahhhC** [comment]
Thanks @KyuaKwon for your nice work! 👍

**Pull request merged**